### PR TITLE
Add WEBP + AVIF Support by default

### LINKED ISSUE #32495: [Issue] Add WEBP + AVIF Support by default
This issue is automatically created based on existing pull request: magento/magento2#32400: Add WEBP + AVIF Support by default

---------
Webp should already be set up by default

### Steps to reproduce:
1. Add webp, avif, etc files to lib/web/images directory.
2. Modify app/code/Magento/Backend/view/adminhtml/layout/admin_login.xml file to use webp, avif, etc file as Magento 2 logo (edit "logo" block "logo_image_src" argument value).
3. Flush cache.
4. Check webp, avif, etc file is used as Magento 2 logo on the admin login page.

### PR DESCRIPTION
**Pull request by in-session**
Webp should already be set up by default

### Steps to reproduce:
1. Add webp, avif, etc files to lib/web/images directory.
2. Modify app/code/Magento/Backend/view/adminhtml/layout/admin_login.xml file to use webp, avif, etc file as Magento 2 logo (edit "logo" block "logo_image_src" argument value).
3. Flush cache.
4. Check webp, avif, etc file is used as Magento 2 logo on the admin login page.

### Resolved issues:
1. [x] resolves magento/magento2#32495: Add WEBP + AVIF Support by default

### CODE REVIEW

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @in-session,
Could you sign Adobe CLA and update the nginx config sample as well?
https://github.com/magento/magento2/blob/2.4-develop/nginx.conf.sample

**Comment by ihor-sviziev** [comment]
Yeah, make sense for sure

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by hostep** [comment]
Maybe also avif sequences? https://github.com/h5bp/server-configs-apache/pull/251

https://github.com/h5bp/server-configs-apache/ is my go to project for verifying apache config settings, maybe double check webp as well? (sorry lack of time here for proper review)

**Comment by in-session** [comment]
Ok let me check i will add also some other like opentype fonts

**Comment by in-session** [comment]
> Maybe also avif sequences? [h5bp/server-configs-apache#251](https://github.com/h5bp/server-configs-apache/pull/251)
> 
> https://github.com/h5bp/server-configs-apache/ is my go to project for verifying apache config settings, maybe double check webp as well? (sorry lack of time here for proper review)

Have you some information about nginx?
`location ~* \.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs`

or

`location ~* \.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs?`

**Comment by ihor-sviziev** [comment]
@in-session I believe it's better to use `location ~* \.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs`  as it will be just simpler to read

**Comment by in-session** [comment]
Ok I think that is good for now

Added Media + Static
- svgz
- webp
- avif
- avifs

Removed Media + Static
- swf

Added Static
- webmanifest
- font/truetype
- font/opentype

Maybe someone can go through the nginx.conf.sample again?

File: pub/static/.htaccess [hunk 1]
```diff
@@ -111,14 +122,17 @@ AddType application/xml xml
     ExpiresByType application/json "access plus 1 year"
 
     # Favicon, images, flash
-    <FilesMatch \.(ico|gif|png|jpg|jpeg|swf|svg)$>
+    <FilesMatch \.(ico|gif|png|jpg|jpeg|svg|svgz|webp|avif|avifs?)$>
```
Comment (ihor-sviziev) [inline pub/static/.htaccess]: I believe the `?` at the end is useless in this case as you listed both `avif` and `avifs` in the list:
```suggestion
    <FilesMatch \.(ico|gif|png|jpg|jpeg|svg|svgz|webp|avif|avifs)$>
```

File: pub/static/.htaccess [hunk 2]
```diff
@@ -72,7 +77,7 @@ AddType application/xml xml
 
 <IfModule mod_headers.c>
 
-    <FilesMatch .*\.(ico|jpg|jpeg|png|gif|svg|js|css|swf|eot|ttf|otf|woff|woff2|html|json)$>
+    <FilesMatch .*\.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs?|js|css|eot|ttf|otf|woff|woff2|html|json|webmanifest)$>
```
Comment (ihor-sviziev) [inline pub/static/.htaccess]:
```suggestion
    <FilesMatch .*\.(ico|jpg|jpeg|png|gif|svg|webp|avif|avifs|js|css|eot|ttf|otf|woff|woff2|html|json|webmanifest)$>
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you check my comments and then squash all your commits to single one?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you squash all your changes to single commit?

**Comment by in-session** [comment]
> Could you squash all your changes to single commit?
It seems like I can't do that.

**Comment by ihor-sviziev** [comment]
@in-session here is a good example of how you can squash all commits: https://www.internalpointers.com/post/squash-commits-into-one-git
Please note that after all your changes, you need to force push your changes.

In case of any questions - please reach me at EngCom Slack.

**Comment by ihor-sviziev** [comment]
Let's leave it not squashed.
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B

**Review by ihor-sviziev — APPROVED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8925](https://jira.corp.magento.com/browse/ENGCOM-8925) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@in-session thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Oscar** [comment]
@engcom-Foxtrot can you please check this PR, it require dev-experience.

**Comment by engcom-Foxtrot** [comment]
QA passed

**Comment by engcom-Oscar** [comment]
@magento create issue

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Pull request merged**